docs/advanced/node-local-storage.md

```diff
@@ -0,0 +1,356 @@
+  kubeconfig      = "${module.controller.kubeconfig}"
+}
+```
+
+For the rest of the example it is assumed the nodes names are:
+`poc-storage-worker-0` and `poc-storage-worker-1`.
+
+Create a Storage Class and a Local Persistent Volume exposing the mounted
+filesystem on each node:
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: local-storage
+provisioner: kubernetes.io/no-provisioner
+volumeBindingMode: WaitForFirstConsumer
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: worker-0-pv
+spec:
+  capacity:
+    storage: 10Gi
+  volumeMode: Filesystem
+  accessModes:
+  - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: local-storage
+  local:
+    path: /mnt/
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - poc-storage-worker-0
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: worker-1-pv
+spec:
+  capacity:
+    storage: 10Gi
+  volumeMode: Filesystem
+  accessModes:
+  - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: local-storage
+  local:
+    path: /mnt/
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - poc-storage-worker-1
+```
+
+The PV should be available:
+
+```sh
+$ kubectl get pv
+NAME          CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS    REASON   AGE
+worker-0-pv   10Gi       RWO            Retain           Available           local-storage            95m
+worker-1-pv   10Gi       RWO            Retain           Available           local-storage            95m
+```
+
+As no PVC was created yet, you should also see no PVC:
+
+```sh
+$ kubectl get pvc
+No resources found.
+```
+
+Create a simple Stateful Set that will use the volumes (via a PVC):
+
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: local-test
+spec:
+  serviceName: "local-service"
+  replicas: 2
+  selector:
+    matchLabels:
+      app: local-test
+  template:
+    metadata:
+      labels:
+        app: local-test
+    spec:
+      containers:
+      - name: test-container
+        image: k8s.gcr.io/busybox
+        command:
+        - "/bin/sh"
+        args:
+        - "-c"
+        - "sleep 100000"
+        volumeMounts:
+        - name: local-vol
+          mountPath: /usr/test-pod
+  volumeClaimTemplates:
+  - metadata:
+      name: local-vol
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      storageClassName: "local-storage"
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+The PVC should be listed and the PV bound:
+
+```sh
+$ kubectl get pods
+NAME           READY   STATUS    RESTARTS   AGE
+local-test-0   1/1     Running   0          54s
+local-test-1   1/1     Running   0          48s
+$ kubectl get pvc
```

Comment:
nit: in a single code block like this if there is a empty line separating multiple commands and it's output it will be helpful.

Reply:
Thanks, fixed!